package.json

```
@@ -35,24 +35,18 @@
         "default": "./lib/index.mjs"
       },
       "require": {
-        "types": "./types/stylelint/index.d.ts",
-        "default": "./lib/index.cjs"
+        "types": "./types/stylelint/index.d.ts"
```

[suggestion] We might need `require.default` to pass `require("stylelint")` (ESM):

```suggestion
        "types": "./types/stylelint/index.d.ts",
        "default": "./lib/index.mjs"
```

---

In addition, it seems necessary to add an additional `default` property to the `PublicApi` type to keep `require("stylelint").default` without any type errors:

```ts
export type PublicApi = PostCSS.PluginCreator<PostcssPluginOptions> & {
  // existing properties...

  /**
   * For CommonJS default import compatibility.
   */
  default: PublicApi;
};
```